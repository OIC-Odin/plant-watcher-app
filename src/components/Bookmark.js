import React, { Component } from 'react';
import { Icon } from 'react-native-elements';
import { Spinner } from './common';

export default class Bookmark extends Component {
  addBookmark() {
    const { addBookmark, articleId } = this.props;
    addBookmark(articleId);
  }

  removeBookmark() {
    const { removeBookmark, articleId } = this.props;
    removeBookmark(articleId);
  }

  render() {
    const { bookmarked, processing } = this.props;
    if (processing) {
      return <Spinner size='small' color='#D04255'/>;
    }
    return (
      <Icon
        type='font-awesome'
        name={bookmarked ? 'bookmark' : 'bookmark-o'}
        color='#D04255'
        size={30}
        onPress={bookmarked ? this.removeBookmark.bind(this) : this.addBookmark.bind(this)}
      />
    );
  }
}
