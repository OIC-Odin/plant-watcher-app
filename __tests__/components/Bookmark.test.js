import React from 'react';
import Bookmark from '../../src/components/Bookmark';
import renderer from 'react-test-renderer';

describe('Bookmark', () => {
  it('snapshot-ロード中', () => {
    const INITIAL_STATE = {
      bookmarked: false,
      addBookmark: () => {},
      removeBookmark: () => {},
      articleId: '-KytE808ZYT-X8YPjwGv',
      processing: true
    };
    const tree = renderer.create(<Bookmark {...INITIAL_STATE} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('snapshot-ブックマークあり', () => {
    const INITIAL_STATE = {
      bookmarked: true,
      addBookmark: () => {},
      removeBookmark: () => {},
      articleId: '-KytE808ZYT-X8YPjwGv',
      processing: false
    };
    const tree = renderer.create(<Bookmark {...INITIAL_STATE} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('snapshot-ブックマークなし', () => {
    const INITIAL_STATE = {
      bookmarked: false,
      addBookmark: () => {},
      removeBookmark: () => {},
      articleId: '-KytE808ZYT-X8YPjwGv',
      processing: false
    };
    const tree = renderer.create(<Bookmark {...INITIAL_STATE} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
