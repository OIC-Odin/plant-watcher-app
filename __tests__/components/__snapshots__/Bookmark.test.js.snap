// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bookmark snapshot-ブックマークあり 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      Array [
        false,
        false,
        false,
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "justifyContent": "center",
        },
        undefined,
      ],
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={false}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#D04255",
          "fontSize": 30,
        },
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ],
        Object {
          "fontFamily": "FontAwesome",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    
  </Text>
</View>
`;

exports[`Bookmark snapshot-ブックマークなし 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      Array [
        false,
        false,
        false,
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "justifyContent": "center",
        },
        undefined,
      ],
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={false}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#D04255",
          "fontSize": 30,
        },
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ],
        Object {
          "fontFamily": "FontAwesome",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    
  </Text>
</View>
`;

exports[`Bookmark snapshot-ロード中 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#D04255"
    hidesWhenStopped={true}
    size="small"
  />
</View>
`;
